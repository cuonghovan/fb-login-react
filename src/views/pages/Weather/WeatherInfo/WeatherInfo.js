import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Title = styled.h3`
  color: purple;
  font-size: 24px;
`;

const Metric = styled.div`
  color: black;
  font-size: 18px;
`;

const MetricContainer = styled.div`
  display: inline-block;
  text-align: left;
`;

const WeatherInfo = ({ data }) => {
  if (Object.keys(data).length === 0 && data.constructor === Object) {
    return null;
  } else {
    return (
      <div>
        <Title>Detailed weather for {data.name}</Title>
        <MetricContainer>
          <Metric>Temperature: {data.main.temp - 273.15}°C</Metric>
          <Metric>Min Temperature: {data.main.temp_min - 273.15}°C</Metric>
          <Metric>Max Temperature: {data.main.temp_max - 273.15}°C</Metric>
          <Metric>Pressure: {data.main.pressure}hpa</Metric>
          <Metric>Humidity: {data.main.humidity}%</Metric>
          <Metric>Wind: {data.wind.speed}m/s</Metric>
        </MetricContainer>
      </div>
    );
  }
};

WeatherInfo.propTypes = {
  data: PropTypes.object
};

export default WeatherInfo;
