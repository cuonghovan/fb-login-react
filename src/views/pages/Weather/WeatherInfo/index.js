export { default as WeatherInfo } from "./WeatherInfo";
export { default } from "./WeatherInfoContainer";
