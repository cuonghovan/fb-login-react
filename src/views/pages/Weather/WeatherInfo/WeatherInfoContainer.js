import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { propsToJS } from "Utils/redux-util";
import { weatherSelectors } from "State/weather";
import WeatherInfo from "./WeatherInfo";

class WeatherInfoContainer extends React.Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    data: PropTypes.object
  };

  render() {
    return <WeatherInfo data={this.props.data} />;
  }
}

const mapStateToProps = state => ({
  data: weatherSelectors.getWeatherInfoSelect(state)
});

export default connect(mapStateToProps, null)(propsToJS(WeatherInfoContainer));
