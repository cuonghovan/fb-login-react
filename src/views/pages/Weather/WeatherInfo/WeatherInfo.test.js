/*global shallow mount*/
import React from "react";
import "jest-styled-components";
import WeatherInfo from "./WeatherInfo";

describe("weather-info", () => {
  const props = {
    data: {
      wind: {
        speed: 4.6,
        deg: 140
      },
      name: "Hanoi",
      sys: {
        type: 1,
        id: 7980,
        message: 0.0054,
        country: "VN",
        sunrise: 1519687115,
        sunset: 1519729216
      },
      dt: 1519722000,
      base: "stations",
      visibility: 6000,
      weather: [
        {
          id: 803,
          main: "Clouds",
          description: "broken clouds",
          icon: "04d"
        }
      ],
      coord: {
        lon: 105.85,
        lat: 21.03
      },
      cod: 200,
      main: {
        temp: 295.15,
        pressure: 1011,
        humidity: 78,
        temp_min: 295.15,
        temp_max: 295.15
      },
      id: 1581130,
      clouds: {
        all: 75
      }
    }
  };

  it("renders correctly", () => {
    const weatherInfo = shallow(<WeatherInfo {...props} />);

    expect(weatherInfo).toMatchSnapshot();
  });

  it("shows prop data", () => {
    const weatherInfo = mount(<WeatherInfo {...props} />);
    let metrics = weatherInfo.find("WeatherInfo__Metric");

    expect(metrics.at(0).text()).toEqual("Temperature: 22°C");
    expect(metrics.at(1).text()).toEqual("Min Temperature: 22°C");
    expect(metrics.at(2).text()).toEqual("Max Temperature: 22°C");
    expect(metrics.at(3).text()).toEqual("Pressure: 1011hpa");
    expect(metrics.at(4).text()).toEqual("Humidity: 78%");
    expect(metrics.at(5).text()).toEqual("Wind: 4.6m/s");
  });
});
