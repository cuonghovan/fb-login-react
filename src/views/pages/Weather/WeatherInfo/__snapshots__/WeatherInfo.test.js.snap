// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather-info renders correctly 1`] = `
<div>
  <WeatherInfo__Title>
    Detailed weather for 
    Hanoi
  </WeatherInfo__Title>
  <WeatherInfo__MetricContainer>
    <WeatherInfo__Metric>
      Temperature: 
      22
      °C
    </WeatherInfo__Metric>
    <WeatherInfo__Metric>
      Min Temperature: 
      22
      °C
    </WeatherInfo__Metric>
    <WeatherInfo__Metric>
      Max Temperature: 
      22
      °C
    </WeatherInfo__Metric>
    <WeatherInfo__Metric>
      Pressure: 
      1011
      hpa
    </WeatherInfo__Metric>
    <WeatherInfo__Metric>
      Humidity: 
      78
      %
    </WeatherInfo__Metric>
    <WeatherInfo__Metric>
      Wind: 
      4.6
      m/s
    </WeatherInfo__Metric>
  </WeatherInfo__MetricContainer>
</div>
`;
